pkg/providers/pricing/pricing.go

```diff
@@ -415,6 +417,23 @@ func (p *Provider) Reset() {
 	p.onDemandPrices = staticPricing
 	// default our spot pricing to the same as the on-demand pricing until a price update
 	p.spotPrices = populateInitialSpotPricing(staticPricing)
-	p.onDemandUpdateTime = initialPriceUpdate
-	p.spotUpdateTime = initialPriceUpdate
+	p.onDemandUpdateTime = getInitialPriceUpdateByRegion(p.region)
+	p.spotUpdateTime = getInitialPriceUpdateByRegion(p.region)
+}
+
+func getInitialPriceUpdateByRegion(region string) time.Time {
```

Comment:
I didn't really love this function and the more I looked at this, the more I was asking the question around why we need the update time in the first place for the generated file data. Why does it matter to us at all? It feels a little clunky to me that we would be using this for any of our logic or our logging if this is just static data that's being leveraged as a fallback.

Looking at pulling it out, it's actually unnecessary that we keep track of the update times. We have some logic that is built around it but those things seem to be easily change-able. I'd opt to just drop keeping track of these update times all together so we can drop this partition mapping that we are doing here.

Reply:
This is also something that I respect to be done in a follow-on. I have a staged commit that I could show you where I dropped it and happy to convert that into a PR if you'd rather merge this as-is withe the drop coming afterwards.

Reply:
Link to the commit of the proposed change: https://github.com/jonathan-innis/karpenter/commit/b3b08bbe019c3466c3905e4c9cd9e9ea06a7990a

Reply:
Took a look at your commit, that looks good to me. I can go either way on putting it in this PR or a follow-up, though since you've already done the work a follow-up seems easiest.